plugins/woocommerce-admin/client/marketplace/components/feedback-modal/feedback-modal.tsx

```diff
@@ -0,0 +1,152 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Modal, Button, TextControl, TextareaControl } from '@wordpress/components';
+import { useDispatch } from '@wordpress/data';
+import { useState, useEffect } from '@wordpress/element';
+import { recordEvent } from '@woocommerce/tracks';
+
+/**
+ * Internal dependencies
+ */
+import './feedback-modal.scss';
+import LikertScale from '../likert-scale/likert-scale';
+
+export default function FeedbackModal(): JSX.Element {
+	const CUSTOMER_EFFORT_SCORE_ACTION = 'marketplace_redesign_2023';
+	const LOCALSTORAGE_KEY_DISMISSAL_COUNT = 'marketplace_redesign_2023_dismissals'; // ensure we don't ask for feedback if the user's already given feedback or declined to multiple times
+	const LOCALSTORAGE_KEY_LAST_REQUESTED_DATE = 'marketplace_redesign_2023_last_shown_date'; // ensure we don't ask for feedback more than once per day
+	const SUPPRESS_IF_DISMISSED_X_TIMES = 1; // if the user dismisses the snackbar this many times, stop asking for feedback
+	const SUPPRESS_IF_AFTER_DATE = '2024-01-01'; // if this date is reached, stop asking for feedback
+
+	// We don't want the "How easy was it to find an extension?" dialog to appear forever:
+	const FEEDBACK_DIALOG_CAN_APPEAR = new Date( SUPPRESS_IF_AFTER_DATE ) > new Date();
+	if( ! FEEDBACK_DIALOG_CAN_APPEAR ) {
```

Comment:
```suggestion
	if ( ! FEEDBACK_DIALOG_CAN_APPEAR ) {
```

Example of a prettier whitespace warning. It's also complaining about lines 57 and 99-101.